src/04 - Web services.md

```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
 - Dates must be encoded as **ISO 8601** strings prior to sending them to the web services.
 - Dates are returned by the web services as **ISO 8601** formatted strings.
 
+# Code list management web services
+
+Address:
+```
+https://{server}:{port}/rms/api/public/noark5/v1/code-lists
+```
+
+## **code-lists**
+
+Gets all code lists supported by the system, the code lists for a given object type, or the code list for a specific field in a given type.
+
+**Request**
+
+``` text
+GET /rms/api/public/noark5/v1/code-lists?type=OBJECT_TYPE&field=FIELD_NAME HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+```
+
+###### Details
+
+- **type** (optional)
+  - type of object to fetch the code lists for
+- **field** (optional)
+  - field in the specified type to fetch the code list for
+
+If **type** and **field** are both not specified, the response will contain all code lists supported by the system.
+
+**Response**
+
+``` text
+Content-Type: application/json
+
+{
+  "results": [
+    {
+      "type": string,
+      "field": string,
+      "values": [
+        {
+          "code": string,
+          "name": string,
+          "description": string,
+          "authority": string
+        },
+        ...
+      ]
+    },
+    ...
+  ]
+}
+```
+
+###### Details
+
+- **results**
+  - array of one or more code lists
+  - **type**
+    - object type that this code list applies to
+  - **field**
+    - field in the object type that this code list applies to
+    - **code**
+      - code or short name of the code list value
+    - **name**
+      - user-friendly name of the code list value
+    - **description**
+      - description of the code list value
+    - **authority** (optional)
+      - authority of the code list value
+
+## **code-lists/{listId}/{code}**
+
+### create and update
+
+Creates or updates a code list value.
+
+**Request**
+
+``` text
+PUT /rms/api/public/noark5/v1/code-lists/{listId}/{code} HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+
+{
+  "name": string,
+  "description": string,
+  "authority": string
+}
+```
+
+###### Details
+
+- **listId**
+  - ID of the code list in which to create or update the code list value
+  - corresponds to a code list as described in - [03 - Code lists](03%20-%20Code%20lists.md)
+- **code**
+  - code or short name of the code list value to create or update
```

Comment:
See my other comment about describing the code as a unique identifier of the code list value.